src/amberscript/parser.cc

```diff
@@ -489,6 +498,15 @@ Result Parser::ParseShaderBlock() {
 
   shader->SetFormat(format);
 
+  token = tokenizer_->PeekNextToken();
+  if (token->IsIdentifier() && token->AsString() == "TARGET_ENV") {
+    tokenizer_->NextToken();
+    token = tokenizer_->NextToken();
+    if (!token->IsIdentifier() && !token->IsString())
+      return Result("expected target environment after TARGET_ENV");
+    shader->SetTargetEnv(token->AsString());
+  }
+
```

Comment:
I think we should probably give an error if we see TARGET_ENV and the shader type is SPIRV-HEX, as the target env parameter won't do anything. We should also change the documentation in this case. @dj2

Reply:
Actually, this is not true if you want to optimize the shader (using spirv-opt). So maybe ignore this or see what Dan thinks we should do.

Reply:
Actually, I am kind of wrong again; spirv-opt with target-env spv1.5 will not change a shader to spv1.5; the SPIR-V version remains unchanged. But we could still allow TARGET_ENV and SPIRV-HEX.

Reply:
ok, so there's nothing I should change here, right?